tools/azure-pipelines/jobs-template.yml

```diff
@@ -132,20 +134,26 @@ jobs:
         restoreKeys: $(CACHE_FALLBACK_KEY)
         path: $(MAVEN_CACHE_FOLDER)
       displayName: Cache Maven local repo
+      continueOnError: true
     
     # download artifacts
     - task: DownloadPipelineArtifact@2
       inputs:
         path: $(CACHE_FLINK_DIR)
         artifact: FlinkCompileCacheDir-${{parameters.stage_name}}
+    - script: |
+        echo "##vso[task.setvariable variable=JAVA_HOME]$(JAVA_HOME_11_X64)"
```

Comment:
Is this to have the tests run on java 11 if java 11 is available on the machine? Shouldn't the container have this stuff set up already (or in other words, what's the benefit in having distinct images if extra steps are required to select the jdk version)?

Reply:
This step is only used on the azure machines (where we do not use the docker image).
That's why I'm changing the java version like this.

Reply:
But ... there is condition for this job(?) that ${{parameters.container}} must be `flink-build-jdk11`. Are we _not_ using `rmetzger/flink-ci:ubuntu-jdk11-amd64-2a765ab` when running on azure machines?

Reply:
Correct, this is a bit ugly. I basically decide whether we are using jdk11 or not based on the container name (that is not used in this context).

I can address this as follows: I build a generic docker image (that does not distinguish between jdk8 / jdk11 because it contains both), and I switch the jdk version for docker and azure machines the same way.

Are you okay if I address this in a follow up? (I can only test this during the night, because I would overwhelm the available machines)